# 1장 2강: 유의수준·검정력·표본 크기의 트레이드오프와 실질적 유의성 — 실습문제

## 실습 목표

- 효과 크기와 유의수준이 같을 때 표본 크기에 따라 검정력이 어떻게 변하는지 설명할 수 있다.
- 목표 검정력과 유의수준을 이용해 필요한 표본 크기를 계산할 수 있다.
- 두 집단의 평균 차이에 대한 Cohen’s d를 계산하고 해석할 수 있다.
- p-value와 효과 크기, 실무 기준을 함께 사용하여 의사결정 근거를 작성할 수 있다.

## 실습 환경 / 데이터

- Python
- NumPy, pandas
- scipy.stats
- statsmodels
- `ames_housing.csv`

주요 컬럼은 다음과 같습니다.

| 컬럼 | 의미 |
|---|---|
| `SalePrice` | 주택 판매가격 |
| `CentralAir` | 중앙 냉방시설 유무(`Y`, `N`) |
| `KitchenQual` | 주방 품질(`Ex`, `Gd`, `TA`, `Fa`) |

> 모든 검정은 별도 지시가 없으면 유의수준 `α = 0.05`를 사용합니다.  
> Cohen’s d는 절댓값을 기준으로 약 0.2는 작은 효과, 0.5는 중간 효과, 0.8 이상은 큰 효과로 해석합니다.

## 실습 준비

1. 필요한 라이브러리를 불러오세요.
2. Ames Housing 데이터를 `df`에 불러오세요.
3. 데이터 크기, 전체 결측치 수, 컬럼명과 상위 5개 행을 확인하세요.

In [5]:
# 실습 준비 코드를 작성하세요.
import numpy as np
import pandas as pd
from scipy import stats
from statsmodels.stats.power import TTestIndPower
import math

df = pd.read_csv('ames_housing.csv')

print(f'데이터 크기 : {df.shape}')
print(f'전체 결측치 수 : {df.isnull().sum().sum()}')
print(f'컬럼 수 : {len(df.columns)}')
df.head()

데이터 크기 : (1460, 10)
전체 결측치 수 : 0
컬럼 수 : 10


,SalePrice,GrLivArea,LotArea,OverallQual,KitchenQual,CentralAir,HeatingQC,PavedDrive,Neighborhood,YearBuilt
0,208500,1710,8450,7,Gd,Y,Ex,Y,CollgCr,2003
1,181500,1262,9600,6,TA,Y,Ex,Y,Veenker,1976
2,223500,1786,11250,7,Gd,Y,Ex,Y,CollgCr,2001
3,140000,1717,9550,7,Gd,Y,Gd,Y,Crawfor,1915
4,250000,2198,14260,8,Gd,Y,Ex,Y,NoRidge,2000


---

## 필수 1. 표본 크기와 검정력 비교

### 문제 1-1. 중앙 냉방시설 비교 연구에 필요한 표본 수 설계

#### 문제 설명

중앙 냉방시설이 있는 주택과 없는 주택의 판매가격을 비교하는 연구를 준비하고 있습니다. 사전 조사에서 표준화 효과 크기는 `0.3` 정도로 예상했습니다.

다음 조건에서 그룹당 표본 크기에 따른 검정력을 비교하고, 목표 검정력 0.8을 확보하기 위해 필요한 표본 수를 계산하세요.

- 예상 효과 크기: `0.3`
- 유의수준: `0.05`
- 양측검정
- 비교할 그룹당 표본 수: `30명`, `100명`, `300명`

#### 요구사항

1. `TTestIndPower()` 객체를 생성하세요.
2. 그룹당 표본 수가 30명, 100명, 300명일 때의 검정력을 각각 계산하세요.
3. 표본 크기별 검정력을 소수점 셋째 자리까지 출력하세요.
4. 유의수준 0.05, 목표 검정력 0.8, 효과 크기 0.3일 때 그룹당 필요한 표본 수를 계산하세요.
5. 계산된 표본 수는 소수점 이하를 올림하여 정수로 출력하세요.
6. 세 표본 크기 중 목표 검정력 0.8을 충족하는 경우를 확인하세요.

#### 해석 질문

**Q1.** 효과 크기와 유의수준이 같을 때 표본 크기가 커지면 검정력은 어떻게 변하나요?  
**Q2.** 검정력이 낮으면 실제 차이가 존재할 때 어떤 오류의 위험이 커지나요?  
**Q3.** 필요한 표본 수를 소수점 이하 올림으로 처리하는 이유는 무엇인가요?  
**Q4.** 표본 수를 무조건 크게 설정하는 것이 항상 최선인가요?

#### 제출 결과

- 표본 크기별 검정력
- 목표 검정력에 필요한 그룹당 표본 수
- 목표 검정력 충족 여부
- Q1~Q4 답변

In [3]:
# # 필수 1 코드를 작성하세요.
# from pathlib import Path
# import numpy as np
# import pandas as pd
# from scipy import stats
# from statsmodels.stats.power import TTestIndPower
# analysis = TTestIndPower()
# effect_size = 0.3
# alpha = 0.05
# target_power = 0.8
# sample_sizes = [30,100,300]

# power_results ={}

# for n in sample_sizes:
#     power= analysis.power(
#         effect_size=effect_size,
#         nobs1=n,
#         alpha=alpha,
#         alternative="two_sided"
#     )
#     power_results[n] = power
#     print(f"그룹당 {n}명일때 검정력 : {power:.3f}")

    


In [10]:
# 필수 1 코드를 작성하세요.
import math
import numpy as np
import pandas as pd
from scipy import stats

# statsmodels가 있으면 그대로 쓰고, 없으면 scipy로 같은 계산을 합니다.
try:
    from statsmodels.stats.power import TTestIndPower
    analysis = TTestIndPower()
    use_sm = True
except ModuleNotFoundError:
    analysis = None
    use_sm = False

def get_power(effect_size, n, alpha):
    """그룹당 표본 수가 n일 때의 검정력(양측검정)"""
    if use_sm:
        return analysis.power(effect_size=effect_size, nobs1=n, alpha=alpha,
                              ratio=1.0, alternative="two-sided")
    dof = 2 * n - 2                         # 자유도
    ncp = effect_size * math.sqrt(n / 2)    # 효과 크기가 만드는 분포 이동량
    t_crit = stats.t.ppf(1 - alpha / 2, dof)
    return stats.nct.sf(t_crit, dof, ncp) + stats.nct.cdf(-t_crit, dof, ncp)

def get_n(effect_size, target_power, alpha):
    """목표 검정력을 채우는 그룹당 표본 수"""
    if use_sm:
        return analysis.solve_power(effect_size=effect_size, power=target_power,
                                    alpha=alpha, ratio=1.0, alternative="two-sided")
    n = 2
    while get_power(effect_size, n, alpha) < target_power:
        n += 1
    return n

# 조건 설정
effect_size = 0.3
alpha = 0.05
target_power = 0.8
sample_sizes = [30, 100, 300]

# 2 & 3. 표본 크기별 검정력  ← 기존 print에 제2종 오류 확률만 추가
print("=== 표본 크기별 검정력 (양측검정) ===")
power_results = {}
for n in sample_sizes:
    power = get_power(effect_size, n, alpha)
    power_results[n] = power
    print(f"그룹당 표본 수 {n:>3}명 : 검정력 = {power:.3f} / 놓칠 확률(제2종 오류) = {1 - power:.3f}")

# 4 & 5. 목표 검정력 0.8에 필요한 표본 수  ← 참고 줄 한 개 추가
required_n = get_n(effect_size, target_power, alpha)
required_n_ceil = math.ceil(required_n)
print("\n=== 목표 검정력 0.8을 위한 표본 크기 ===")
print(f"계산된 필요 표본 수(소수) : {required_n:.4f}")
print(f"최종 필요 표본 수(올림)   : 그룹당 {required_n_ceil}명")
print(f"참고) {required_n_ceil - 1}명 검정력 {get_power(effect_size, required_n_ceil - 1, alpha):.3f} (0.8 미달) / "
      f"{required_n_ceil}명 검정력 {get_power(effect_size, required_n_ceil, alpha):.3f} (충족)")

# 6. 목표 검정력 충족 여부
print("\n=== 목표 검정력(0.8) 충족 여부 ===")
for n in sample_sizes:
    status = "충족" if power_results[n] >= target_power else "미충족"
    print(f"그룹당 {n:>3}명 : {status} (검정력 {power_results[n]:.3f})")


=== 표본 크기별 검정력 (양측검정) ===
그룹당 표본 수  30명 : 검정력 = 0.208 / 놓칠 확률(제2종 오류) = 0.792
그룹당 표본 수 100명 : 검정력 = 0.560 / 놓칠 확률(제2종 오류) = 0.440
그룹당 표본 수 300명 : 검정력 = 0.956 / 놓칠 확률(제2종 오류) = 0.044

=== 목표 검정력 0.8을 위한 표본 크기 ===
계산된 필요 표본 수(소수) : 175.3847
최종 필요 표본 수(올림)   : 그룹당 176명
참고) 175명 검정력 0.799 (0.8 미달) / 176명 검정력 0.801 (충족)

=== 목표 검정력(0.8) 충족 여부 ===
그룹당  30명 : 미충족 (검정력 0.208)
그룹당 100명 : 미충족 (검정력 0.560)
그룹당 300명 : 충족 (검정력 0.956)


### 필수 1 답변 작성란

**Q1.** 효과 크기와 유의수준이 같을 때 표본 크기가 커지면 검정력은 어떻게 변하나요?  
-> 표본 크기가 커질수록 일반적으로 검정력이 높아진다.
-> 표본이 많아질수록 실제 차이를 우연한 변동과 구분하기 쉬워짐

**Q2.** 검정력이 낮으면 실제 차이가 존재할 때 어떤 오류의 위험이 커지나요?  
-> 실제 차이가 있는데도 귀무가설을 기각하지 못하는 제 2종 오류의 위험이 커짐
-> 예를 들어 검정력이 0.8이라면 가정한 실제 차이를 놓칠 확률인 'β'는 '1-0.8=0.2'입니다.

**Q3.** 필요한 표본 수를 소수점 이하 올림으로 처리하는 이유는 무엇인가요?  
-> 사람이나 주택을 소수 단위로 모집할 수 없고
-> 내림하면 계산된 목표 검정력을 충족하지 못할 수 있다. 

**Q4.** 표본 수를 무조건 크게 설정하는 것이 항상 최선인가요?
-> 아닙니다. 표본 확보에는 비용과 시간이 소요가 됨

---

## 필수 2. 통계적 유의성과 실질적 유의성 종합 판단

### 문제 2-1. 중앙 냉방시설 유무에 따른 판매가격 차이 분석

#### 문제 설명

한 부동산 회사는 중앙 냉방시설이 있는 주택과 없는 주택의 평균 판매가격 차이를 분석하려고 합니다. 회사는 두 집단의 평균 판매가격 차이가 **100,000달러 이상**이어야 냉방시설 설치 지원 사업을 검토할 실무적 가치가 있다고 정했습니다.

#### 요구사항

1. `CentralAir == "Y"`인 주택의 `SalePrice`를 `air_yes`에 저장하세요.
2. `CentralAir == "N"`인 주택의 `SalePrice`를 `air_no`에 저장하세요.
3. 두 집단의 표본 수와 평균 판매가격을 출력하세요.
4. `stats.ttest_ind()`로 두 집단 평균 차이에 대한 양측검정을 수행하세요.
5. 제공된 공식에 따라 Cohen’s d 계산 함수를 작성하고 효과 크기를 구하세요.
6. 평균 차이, p-value, Cohen’s d를 출력하세요.
7. 다음 기준으로 결과를 판단하세요.
   - `p-value ≤ 0.05`: 통계적으로 유의함
   - `|Cohen’s d| ≥ 0.8`: 큰 효과
   - `|평균 차이| ≥ 100000`: 회사 기준에서 실질적으로 유의함
8. 세 결과를 종합하여 사업 검토 여부에 대한 근거를 작성하세요.

#### Cohen’s d 공식

두 집단의 평균 차이를 합동 표준편차로 나누어 계산합니다.

$$
d = \frac{\bar{x}_1 - \bar{x}_2}{s_p}
$$

$$
s_p = \sqrt{\frac{(n_1-1)s_1^2 + (n_2-1)s_2^2}{n_1+n_2-2}}
$$

- $\bar{x}_1$, $\bar{x}_2$: 각 집단의 평균
- $s_1$, $s_2$: 각 집단의 표본 표준편차
- $n_1$, $n_2$: 각 집단의 표본 수
- $s_p$: 합동 표준편차

#### 해석 질문

**Q1.** p-value와 Cohen’s d는 각각 어떤 정보를 제공하나요?  
**Q2.** 두 집단의 판매가격 차이는 통계적으로 유의한가요?  
**Q3.** Cohen’s d를 기준으로 효과 크기는 어느 정도인가요?  
**Q4.** 회사가 정한 100,000달러 기준에서 실질적으로 유의한가요?  
**Q5.** 통계적으로 유의하고 효과 크기가 크더라도 회사의 실무 기준을 충족하지 못할 수 있나요?

#### 제출 결과

- 집단별 표본 수와 평균
- 독립표본 t검정 결과
- 평균 차이와 Cohen’s d
- 통계적·효과크기·회사 기준 판단
- 최종 의사결정 근거
- Q1~Q5 답변

In [11]:
# 회사 기준으로는 10만달러로 기준이었는데 기준을 충족하지 못하였음. 
# 최종판안 : 일부 통계적 근거는 있지만 회사의 모든 기준을 충족하지는 않습니다.

# 필수 2 코드를 작성하세요.
import math
import pandas as pd
from scipy import stats

# 1 & 2. 집단 나누기
air_yes = df[df["CentralAir"] == "Y"]["SalePrice"]
air_no = df[df["CentralAir"] == "N"]["SalePrice"]

# 3. 표본 수와 평균
print("=== 집단별 표본 수와 평균 ===")
print(f"냉방시설 있음(Y) : {len(air_yes)}채, 평균 {air_yes.mean():,.1f}달러")
print(f"냉방시설 없음(N) : {len(air_no)}채, 평균 {air_no.mean():,.1f}달러")

# 4. 독립표본 t검정 (양측)
t_stat, p_value = stats.ttest_ind(air_yes, air_no, equal_var=True)

# 5. Cohen's d 계산 함수
def cohen_d(group1, group2):
    n1, n2 = len(group1), len(group2)
    s1, s2 = group1.std(ddof=1), group2.std(ddof=1)
    sp = math.sqrt(((n1 - 1) * s1 ** 2 + (n2 - 1) * s2 ** 2) / (n1 + n2 - 2))
    return (group1.mean() - group2.mean()) / sp

mean_diff = air_yes.mean() - air_no.mean()
d = cohen_d(air_yes, air_no)

# 6. 결과 출력
print("\n=== 검정 결과 ===")
print(f"t통계량   : {t_stat:.3f}")
print(f"p-value   : {p_value:.3e}")
print(f"평균 차이 : {mean_diff:,.1f}달러")
print(f"Cohen's d : {d:.3f}")

# 7. 세 가지 기준 판단
alpha = 0.05
d_cut = 0.8
money_cut = 100000

stat_sig = p_value <= alpha
big_effect = abs(d) >= d_cut
practical = abs(mean_diff) >= money_cut

print("\n=== 기준별 판단 ===")
print(f"통계적 유의성 (p <= {alpha})        : {'충족' if stat_sig else '미충족'}")
print(f"큰 효과 (|d| >= {d_cut})             : {'충족' if big_effect else '미충족'}")
print(f"실질적 유의성 (|차이| >= {money_cut:,}) : {'충족' if practical else '미충족'}")

# 8. 종합 의사결정 근거
print("\n=== 최종 의사결정 근거 ===")
if stat_sig and big_effect and practical:
    print("세 기준을 모두 충족하므로 냉방시설 설치 지원 사업을 검토할 근거가 충분합니다.")
else:
    print(f"통계적으로는 유의하고(p={p_value:.3e}) 효과 크기도 크지만(d={d:.3f}),")
    print(f"평균 차이 {mean_diff:,.1f}달러가 회사 기준 {money_cut:,}달러에 "
          f"{money_cut - abs(mean_diff):,.1f}달러 부족합니다.")
    print("따라서 통계적 근거는 있으나 회사의 실무 기준은 충족하지 못해 사업 검토는 보류가 타당합니다.")


=== 집단별 표본 수와 평균 ===
냉방시설 있음(Y) : 1365채, 평균 186,186.7달러
냉방시설 없음(N) : 95채, 평균 105,264.1달러

=== 검정 결과 ===
t통계량   : 9.915
p-value   : 1.810e-22
평균 차이 : 80,922.6달러
Cohen's d : 1.052

=== 기준별 판단 ===
통계적 유의성 (p <= 0.05)        : 충족
큰 효과 (|d| >= 0.8)             : 충족
실질적 유의성 (|차이| >= 100,000) : 미충족

=== 최종 의사결정 근거 ===
통계적으로는 유의하고(p=1.810e-22) 효과 크기도 크지만(d=1.052),
평균 차이 80,922.6달러가 회사 기준 100,000달러에 19,077.4달러 부족합니다.
따라서 통계적 근거는 있으나 회사의 실무 기준은 충족하지 못해 사업 검토는 보류가 타당합니다.


### 필수 2 답변 작성란

**Q1.** p-value와 Cohen’s d는 각각 어떤 정보를 제공하나요?  
-> p-value는 관측된 차이가 통계적으로 유의한지 판단하는데 사용
-> Cohen's d는 두 집단 차이의 크기를 표준화하여 보여줌

**Q2.** 두 집단의 판매가격 차이는 통계적으로 유의한가요?  
-> 유의함. p-value가 0.05보다 작음
-> 귀무가설 : 냉방 시설 유무는 주택가격 평균에 차이를 보이지 않는다. -> 기각


**Q3.** Cohen’s d를 기준으로 효과 크기는 어느 정도인가요?  
-> Cohen’s의 절대값이 1.05이고 규정한 0.8보다 크기 때문에 큰 효과로 해석될 수 있다.

**Q4.** 회사가 정한 100,000달러 기준에서 실질적으로 유의한가요?  
-> 유의하지 않음. 80,900달러 정도였기 때문에 회사의 기대 미충족

**Q5.** 통계적으로 유의하고 효과 크기가 크더라도 회사의 실무 기준을 충족하지 못할 수 있나요?
-> 그렇다. 
-> 통계적 유의성은 차이가 있다는 근거가 있는가?
-> 효과 크기는 자료의 흩어짐에 비해 차이가 큰가?
-> 회사 입장은 업무에서 요구하는 최소 금액 차이를 넘는가?



---

## 과제. 주방 품질에 따른 판매가격 차이 분석

### 문제 3-1. 주방 품질이 `Gd`인 집단과 `TA`인 집단 비교

#### 문제 설명

부동산 중개회사는 주방 품질이 `Gd`(Good)인 주택과 `TA`(Typical/Average)인 주택의 평균 판매가격을 비교하려고 합니다. 회사는 평균 판매가격 차이가 **50,000달러 이상**이면 마케팅에서 주방 품질의 차이를 강조할 실무적 가치가 있다고 판단합니다.

> 필수 2에서 학습한 검정과 효과 크기 계산 절차를 새로운 두 집단에 적용하는 과제입니다.

#### 요구사항

1. 주방 품질이 `Gd`인 집단과 `TA`인 집단의 `SalePrice`를 각각 준비하세요.
2. 두 집단의 표본 수와 평균 판매가격을 출력하세요.
3. 두 집단의 평균 차이에 대한 양측 독립표본 t검정을 수행하세요.
4. Cohen’s d를 계산하세요.
5. 평균 차이, p-value, Cohen’s d를 출력하세요.
6. 다음 기준에 따라 각각 판단하세요.
   - 통계적 유의성: `p-value ≤ 0.05`
   - 큰 효과: `|Cohen’s d| ≥ 0.8`
   - 실무적 유의성: `|평균 차이| ≥ 50000`
7. 세 가지 판단을 종합하여 주방 품질을 마케팅에서 강조할 근거가 있는지 결론을 작성하세요.

#### 해석 질문

**Q1.** 두 집단의 판매가격 차이는 통계적으로 유의한가요?  
**Q2.** Cohen’s d를 기준으로 효과 크기는 어느 정도인가요?  
**Q3.** 회사가 정한 50,000달러 기준을 충족하나요?  
**Q4.** 최종적으로 주방 품질 차이를 마케팅에서 강조할 근거가 있다고 볼 수 있나요?

#### 제출 결과

- 집단별 표본 수와 평균
- t통계량과 p-value
- 평균 차이와 Cohen’s d
- 통계적·실질적 유의성 판단
- 최종 결론
- Q1~Q4 답변

In [13]:
# 과제 코드를 작성하세요.

import math
import pandas as pd
from scipy import stats

def cohen_d(group1, group2):
    n1, n2 = len(group1), len(group2)
    s1, s2 = group1.std(ddof=1), group2.std(ddof=1)
    sp = math.sqrt(((n1 - 1) * s1 ** 2 + (n2 - 1) * s2 ** 2) / (n1 + n2 - 2))
    return (group1.mean() - group2.mean()) / sp

# 1. 두 집단 준비
gd_price = df[df["KitchenQual"] == "Gd"]["SalePrice"]
ta_price = df[df["KitchenQual"] == "TA"]["SalePrice"]

# 2. 표본 수와 평균
print("=== 집단별 표본 수와 평균 ===")
print(f"주방 품질 Gd : {len(gd_price)}채, 평균 {gd_price.mean():,.1f}달러")
print(f"주방 품질 TA : {len(ta_price)}채, 평균 {ta_price.mean():,.1f}달러")

# 3. 독립표본 t검정 (양측)
t_stat, p_value = stats.ttest_ind(gd_price, ta_price, equal_var=True)

# 4 & 5. 평균 차이, Cohen's d 출력
mean_diff = gd_price.mean() - ta_price.mean()
d = cohen_d(gd_price, ta_price)

print("\n=== 검정 결과 ===")
print(f"t통계량   : {t_stat:.3f}")
print(f"p-value   : {p_value:.3e}")
print(f"평균 차이 : {mean_diff:,.1f}달러")
print(f"Cohen's d : {d:.3f}")

# 6. 기준별 판단
alpha = 0.05
d_cut = 0.8
money_cut = 50000

stat_sig = p_value <= alpha
big_effect = abs(d) >= d_cut
practical = abs(mean_diff) >= money_cut

print("\n=== 기준별 판단 ===")
print(f"통계적 유의성 (p <= {alpha})       : {'충족' if stat_sig else '미충족'}")
print(f"큰 효과 (|d| >= {d_cut})            : {'충족' if big_effect else '미충족'}")
print(f"실질적 유의성 (|차이| >= {money_cut:,}) : {'충족' if practical else '미충족'}")

# 7. 최종 결론
print("\n=== 최종 결론 ===")
if stat_sig and big_effect and practical:
    print(f"세 기준을 모두 충족합니다. 주방 품질이 Gd인 주택이 TA보다 평균 "
          f"{mean_diff:,.1f}달러 비싸고(p={p_value:.3e}, d={d:.3f}),")
    print(f"회사 기준 {money_cut:,}달러도 넘으므로 주방 품질 차이를 마케팅에서 강조할 근거가 있습니다.")
else:
    print("세 기준 중 일부를 충족하지 못하므로 마케팅에서 강조하기에는 근거가 부족합니다.")


=== 집단별 표본 수와 평균 ===
주방 품질 Gd : 586채, 평균 212,116.0달러
주방 품질 TA : 735채, 평균 139,962.5달러

=== 검정 결과 ===
t통계량   : 25.263
p-value   : 3.557e-115
평균 차이 : 72,153.5달러
Cohen's d : 1.399

=== 기준별 판단 ===
통계적 유의성 (p <= 0.05)       : 충족
큰 효과 (|d| >= 0.8)            : 충족
실질적 유의성 (|차이| >= 50,000) : 충족

=== 최종 결론 ===
세 기준을 모두 충족합니다. 주방 품질이 Gd인 주택이 TA보다 평균 72,153.5달러 비싸고(p=3.557e-115, d=1.399),
회사 기준 50,000달러도 넘으므로 주방 품질 차이를 마케팅에서 강조할 근거가 있습니다.


### 과제 답변 작성란

**Q1.** 두 집단의 판매가격 차이는 통계적으로 유의한가요?  
-> 유의합니다. p-value가 3.557e-115로 유의수준 0.05보다 작으므로 귀무가설을 기각합니다.

**Q2.** Cohen’s d를 기준으로 효과 크기는 어느 정도인가요?  
-> Cohen's d가 1.399로 0.8보다 크므로 큰 효과 기준이라고 볼 수 있다. 

**Q3.** 회사가 정한 50,000달러 기준을 충족하나요?  
-> 충족합니다. Gd : 212,116.0달러, TA : 139,962.5달러로 평균 차이는 72,153.5달러입니다. 회사가 정한 50,000달러보다 더 높습니다.

**Q4.** 최종적으로 주방 품질 차이를 마케팅에서 강조할 근거가 있다고 볼 수 있나요?
-> 강조할 근거가 있습니다. 통계적 유의성(p < 0.05), 큰 효과(|d| ≥ 0.8), 실무 기준(평균 차이 ≥ 50,000달러) 세 가지를 모두 충족됩니다.

---

## 실습 마무리

1. 검정력은 무엇이며 제2종 오류와 어떤 관계가 있나요?
--> 검정력 : 실제로 차이가 있을 때 그 차이를 통계적으로 찾아낼 확률
--> 제2종 오류 : 실제 차이가 있는데도 귀무가설을 기각하지 못하고 놓치는 오류
--> 검정력 = 1 − 제 2종 오류

2. 같은 효과 크기와 유의수준에서 표본 크기가 커지면 검정력은 어떻게 변하나요?
--> 검정력이 높아집니다. 효과크기와 유의수준을 똑같이 두고 표본만 늘렸을 때 
--> 평균 추정이 정밀해져서 실제 차이와 변동을 구분하기 쉬워지기 때문입니다.

3. 목표 검정력이 높거나 발견하려는 효과가 작을수록 필요한 표본 수는 어떻게 변하나요?
-->표본 수를 늘립니다. 검정력을 높이거나, 더 작은 차이까지 찾아내려고 할수록 표본 수를 더 많이 모아야 합니다.

4. p-value와 Cohen’s d를 함께 확인해야 하는 이유는 무엇인가요?
--> p-value : 이 차이가 우연인가? 
--> Cohen's d : 그 차이가 얼마나 큰가?
--> p-value는 표본 수에 따라 결과가 달라질 수 있고,  d만 보면 그 차이가 우연인지 판단할 수 없습니다.  

5. 통계적으로 유의한 결과가 반드시 실무적으로 중요한 결과를 의미하나요?
--> 아니요. 필수 2를 예시로 p=1.810e-22로 통계적으로 유의하고 Cohen's d : 1.052, 평균 차이가 80,922.6달러였지만,
--> 회사가 정한 10만 달러 기준에 19,077.4달러 부족해서 사업 검토를 보류하는 것이 타당했다.
--> 주방 품질은 평균 차이 72,153.5달러로 회사 기준 5만 달러를 넘어 세 조건을 모두 충족했던 결과를 보면
--> 통계적 유의성과 효과 크기는 차이가 있다는 근거이지 실무에서는 회사의 기준이나 비용 등 여러 요인을 따져 결정된다.